server/store/sqlstore/store.go

```diff
@@ -981,6 +995,81 @@ func (s *SQLStore) IsUserPresentInWhitelist(userID string) (bool, error) {
 	return result != "", nil
 }
 
+func (s *SQLStore) StoreInvitedUser(invitedUser *storemodels.InvitedUser) error {
+	pendingSince := invitedUser.InvitePendingSince.UnixMicro()
+	lastSentAt := invitedUser.InviteLastSentAt.UnixMicro()
+
+	query := s.getQueryBuilder().
+		Insert(invitedUsersTableName).
+		Columns("mmUserID", "invitePendingSince", "inviteLastSentAt").
+		Values(invitedUser.ID, pendingSince, lastSentAt).
+		SuffixExpr(sq.Expr("ON CONFLICT (mmUserID) DO UPDATE SET invitePendingSince = ?, inviteLastSentAt = ?", pendingSince, lastSentAt))
+
+	if _, err := query.Exec(); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func (s *SQLStore) GetInvitedUser(mmUserID string) (*storemodels.InvitedUser, error) {
+	query := s.getQueryBuilder().
+		Select("mmUserID", "invitePendingSince", "inviteLastSentAt").
+		From(invitedUsersTableName).
+		Where(sq.Eq{"mmUserID": mmUserID})
+
+	rows, err := query.Query()
+	if err != nil {
+		return nil, err
+	}
+	defer rows.Close()
+
+	var result *storemodels.InvitedUser
+	if rows.Next() {
+		var id string
+		var pendingSince int64
+		var lastSentAt int64
+
+		if scanErr := rows.Scan(&id, &pendingSince, &lastSentAt); scanErr != nil {
+			return nil, scanErr
+		}
+
+		result = &storemodels.InvitedUser{
+			ID:                 id,
+			InvitePendingSince: time.UnixMicro(pendingSince),
+			InviteLastSentAt:   time.UnixMicro(pendingSince),
+		}
+	}
+
+	return result, nil
+}
+
+func (s *SQLStore) DeleteUserInvite(mmUserID string) error {
+	if _, err := s.getQueryBuilder().Delete(invitedUsersTableName).Where(sq.Eq{"mmUserID": mmUserID}).Exec(); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func (s *SQLStore) GetSizeOfInvitedUsers() (int, error) {
```

Comment:
nit: `size` vs. `count`, might be easier to understand as `GetPendingUserInvitesCount()` or some such.

Reply:
Agreed, but copied from `GetSizeOfWhitelist` 1/5? 4/5?

Reply:
-1/5 for now, something to go through and fix en masse later :)